.github/workflows/build_binaries.yml

```diff
@@ -0,0 +1,65 @@
+name: Build Binaries # Continuous Deployment
+
+permissions:
+  contents: write
+
+on:
+  release:
+    types: [published]
+
+env:
+  CARGO_INCREMENTAL: 0
+  CARGO_NET_GIT_FETCH_WITH_CLI: true
+  CARGO_NET_RETRY: 10
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: 1
+  RUSTFLAGS: -D warnings
+  RUSTUP_MAX_RETRIES: 10
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  upload-assets:
+    name: ${{ matrix.target }}
+    if: github.repository_owner == 'shell-pool' && startsWith(github.event.release.name, 'shpool')
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - target: aarch64-unknown-linux-gnu
+            os: ubuntu-22.04
+          - target: aarch64-unknown-linux-musl
+            os: ubuntu-22.04
+          - target: x86_64-unknown-linux-gnu
+            os: ubuntu-22.04
+          - target: x86_64-unknown-linux-musl
+            os: ubuntu-22.04
+          #- target: aarch64-apple-darwin
+          #  os: macos-12
+          #- target: x86_64-apple-darwin
+          #  os: macos-12
+          #- target: x86_64-unknown-freebsd
+          #  os: ubuntu-22.04
+    timeout-minutes: 60
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
```

Comment:
rather than using @tag we are supposed to use a full commit hash for security reasons. Somewhat annoying

Reply:
Done. Hopefully with dependabot configured, this will become more manageable.